webcomponents/tool/src/main/frontend/js/calendar/sakai-calendar.js

```diff
@@ -60,58 +60,67 @@ export class SakaiCalendar extends LionCalendar {
   static get properties() {
 
     return {
-      i18n: { attribute: false, type: Object },
-      siteId: { attribute: "site-id", type: String },
       userId: { attribute: "user-id", type: String },
-      selectedDate: { attribute: false, type: Date },
-      events: { attribute: false, type: Array },
-      days: { attribute: false, type: Number },
+      siteId: { attribute: "site-id", type: String },
+      _i18n: { attribute: false, type: Object },
+      _selectedDate: { attribute: false, type: Date },
+      _events: { attribute: false, type: Array },
+      _days: { attribute: false, type: Number },
     };
   }
 
   constructor() {
 
     super();
-    loadProperties("calendar").then(r => this.i18n = r);
-    this.daysEvents = [];
+
+    this._daysEvents = [];
 
     this.addEventListener("user-selected-date-changed", event => {
 
       const time = event.detail.selectedDate.getTime();
-      this.daysEvents = this.events.filter(e => e.start > time && e.start < (time + 24 * 60 * 60 * 1000));
-      this.selectedDate = event.detail.selectedDate;
+      this._daysEvents = this._events.filter(e => e.start > time && e.start < (time + 24 * 60 * 60 * 1000));
+      this._selectedDate = event.detail.selectedDate;
     });
-  }
 
-  set siteId(value) {
+    // This is a hack. There's something about the way Lion's calendar connects and renders which
+    // means we can't wait for this._i18n in shouldUpdate. So, we add placeholders in the _i18n
+    // object. Nasty. If I can work it out, I'll come back and fix this.
+    //TODO: Fix this properly
+    this._i18n = { days_message: "", events_for: "", today: "" };
```

Comment:
The error should not show up because of Lion's calendar IMO.  If that were the case we wouldn't have similar issues elsewhere in web components. Ex: https://sakaiproject.atlassian.net/browse/SAK-49235

Reply:
It's all about making i18n a reactive property and waiting until it is present before attempting to render.